x-pack/plugins/viz_editor/pie_chart_plugin/pie_chart_plugin.tsx

```diff
@@ -129,11 +132,71 @@ function getSuggestion(visModel: PieChartVisModel) {
   };
 }
 
+function getSuggestionsForField(datasourceRef: string, field: Field, visModel: PieChartVisModel) {
+  const operationNames = getOperationsForField(field);
+
+  if (operationNames.length === 0) {
+    return [];
+  }
+
+  let firstOperation: SelectOperation;
```

Comment:
A (subjective) nit pick: I personally find that `let` statements add mental overhead that can often be avoided. When I see a `let`, I now need to be careful to follow the code to see how it's mutated over the course of the function. This could be a `const` with a bit of refactoring:

```js
function defaultOperation(field) {
  const [operationName] = getOperationsForField(field);

  if (!operationName) {
    return;
  }

  switch (operationName) {
    case 'terms':
      return {
        operation: 'terms',
        argument: { field: field.name, size: 5 },
      };
    case 'date_histogram':
      return {
        operation: 'date_histogram',
        argument: { field: field.name, interval: '1d' },
      };
    case 'count':
      return { operation: 'count' };
    default:
      return {
        operation: operationName,
        argument: { field: field.name },
      };
  }
}


function getSuggestionsForField(/* etc */) {
  const firstOperation = defaultOperation(field);

  if (!firstOperation) {
    return [];
  }

  // etc
}
```

Reply:
This is a good suggestion, and I am going to clarify the logic today. It's sort of out of scope for this PR to even touch the suggestions, I will split it out.